pkg/actors/reminders/reminders.go

```diff
@@ -0,0 +1,975 @@
+	r.resiliency = resiliency
+}
+
+func (r *reminders) SetLookupActorFn(fn internal.LookupActorFn) {
+	r.lookUpActorFn = fn
+}
+
+func (r *reminders) DrainRebalancedReminders(actorType string, actorID string, actorKey string) {
+	r.remindersLock.RLock()
+	reminders := r.reminders[actorType]
+	r.remindersLock.RUnlock()
+
+	for _, rem := range reminders {
+		// r.reminder refers to the actual reminder struct that is saved in the db
+		if rem.Reminder.ActorType == actorType && rem.Reminder.ActorID == "actorID" {
+			reminderKey := constructCompositeKey(actorKey, rem.Reminder.Name)
+			stopChan, exists := r.activeReminders.Load(reminderKey)
+			if exists {
+				close(stopChan.(chan struct{}))
+				r.activeReminders.Delete(reminderKey)
+			}
+		}
+	}
+}
+
+func (r *reminders) CreateReminder(ctx context.Context, reminder *internal.Reminder) error {
+	store, err := r.stateStoreProviderFn()
+	if err != nil {
+		return err
+	}
+
+	if !r.waitForEvaluationChan() {
+		return errors.New("error creating reminder: timed out after 5s")
+	}
+
+	r.remindersStoringLock.Lock()
+	defer r.remindersStoringLock.Unlock()
+
+	existing, ok := r.getReminder(reminder.Name, reminder.ActorType, reminder.ActorID)
+	if ok {
+		if existing.RequiresUpdating(reminder) {
+			err = r.doDeleteReminder(ctx, reminder.ActorType, reminder.ActorID, reminder.Name)
+			if err != nil {
+				return err
+			}
+		} else {
+			return nil
+		}
+	}
+
+	stop := make(chan struct{})
+
+	err = r.storeReminder(ctx, store, reminder, stop)
+	if err != nil {
+		return fmt.Errorf("error storing reminder: %w", err)
+	}
+	return r.startReminder(reminder, stop)
+}
+
+func (r *reminders) Close() error {
+	// Close the runningCh
+	close(r.runningCh)
+	return nil
+}
+
+func (r *reminders) Init(ctx context.Context) {
+	r.evaluateReminders(ctx)
+}
+
+func (r *reminders) GetReminder(ctx context.Context, req *internal.GetReminderRequest) (*internal.Reminder, error) {
+	list, _, err := r.getRemindersForActorType(ctx, req.ActorType, false)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range list {
+		if r.Reminder.ActorID == req.ActorID && r.Reminder.Name == req.Name {
+			return &internal.Reminder{
+				Data:    r.Reminder.Data,
+				DueTime: r.Reminder.DueTime,
+				Period:  r.Reminder.Period,
+			}, nil
+		}
+	}
+	return nil, nil
+}
+
+func (r *reminders) DeleteReminder(ctx context.Context, req internal.DeleteReminderRequest) error {
+	if !r.waitForEvaluationChan() {
+		return errors.New("error deleting reminder: timed out after 5s")
+	}
+
+	r.remindersStoringLock.Lock()
+	defer r.remindersStoringLock.Unlock()
+
+	return r.doDeleteReminder(ctx, req.ActorType, req.ActorID, req.Name)
+}
+
+func (r *reminders) RenameReminder(ctx context.Context, req *internal.RenameReminderRequest) error {
+	log.Warn("[DEPRECATION NOTICE] Currently RenameReminder renames by deleting-then-inserting-again. This implementation is not fault-tolerant, as a failed insert after deletion would result in no reminder")
```

Comment:
Looks like this was deprecated almost 1 year ago in Dapr 1.9 (#4957). Is it time we remove this method maybe?
CC: @dapr/maintainers-dapr

Reply:
Yes, lets remove this

Reply:
I've opened #6711 to track this.